File: VB/Form1.vb
Imports System
Imports System.Drawing.Design
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            ' Create an End-User Designer form.
            Dim designForm As New XRDesignForm()

            ' Handle the DesignPanelLoaded event.
            AddHandler designForm.DesignMdiController.DesignPanelLoaded, AddressOf DesignMdiController_DesignPanelLoaded
            ' Load a report into the Designer.
            designForm.OpenReport(New XtraReport1())

            ' Show the End-User Designer form, modally.
            designForm.ShowDialog()

        End Sub

        Private Sub DesignMdiController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            ' Get the Toolbox service from the Designer host.
            Dim ts As IToolboxService = DirectCast(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)

            ' Add a new category to the Toolbox,
            ' and place a custom control into it.
            Dim item As New ToolboxItem(GetType(ProgressBar))
            ts.AddToolboxItem(item, "New Category")
        End Sub
    End Class
End Namespace

File: CS/Form1.cs
using System;
using System.Drawing.Design;
using System.Windows.Forms;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // Create an End-User Designer form.
            XRDesignForm designForm = new XRDesignForm();

            // Handle the DesignPanelLoaded event.
            designForm.DesignMdiController.DesignPanelLoaded +=
                DesignMdiController_DesignPanelLoaded;            
            // Load a report into the Designer.
            designForm.OpenReport(new XtraReport1());

            // Show the End-User Designer form, modally.
            designForm.ShowDialog();

        }

        void DesignMdiController_DesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            // Get the Toolbox service from the Designer host.
            IToolboxService ts =
                (IToolboxService)e.DesignerHost.GetService(typeof(IToolboxService));

            // Add a new category to the Toolbox,
            // and place a custom control into it.
            ToolboxItem item = new ToolboxItem(typeof(ProgressBar));
            ts.AddToolboxItem(item, "New Category");
        }
    }
}
